subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -233,8 +233,23 @@ img_mgmt_slot_in_use(int slot)
 	int active_slot = img_mgmt_active_slot(image);
 
 #if !defined(CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP)
-	if (slot == img_mgmt_get_next_boot_slot(image, NULL)) {
+	enum img_mgmt_next_boot_type type;
+	int nbs = img_mgmt_get_next_boot_slot(image, &type);
+
+	if (slot == nbs && type == NEXT_BOOT_TYPE_REVERT) {
+		LOG_DBG("(%d) Refused erase revert", slot);
+		return 1;
+	}
+
+	if ((slot == nbs && type == NEXT_BOOT_TYPE_TEST) ||
+	    (active_slot != nbs && type == NEXT_BOOT_TYPE_NORMAL)) {
+#if defined(CONFIG_MCUMGR_GRP_IMG_ALLOW_ERASE_PENDING)
+		LOG_DBG("(%d) Allowed erase pending", slot);
+		return 0;
```

Comment:
`return (active_slot == slot)`
Otherwise you may allow to erase the currently running slot if the secondary slot marked as permanent swap.,

Reply:
second check will be more legible if current boot type would be used:
slot == nbs && curr_type == BOOT_TYPE_TEST